nat-lab/tests/utils/batching.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+import math
+import os
+import pytest
+import subprocess
+import typing
+from scapy.all import PcapReader, Packet  # type: ignore
+from typing import Callable, List, Optional
+
+
+def generate_histogram(data: list[int], buckets: int, bucket_size: int) -> List[int]:
+    if len(data) == 0:
+        return []
+
+    sorted_values = sorted(data)
+    max_val = sorted_values[-1]
```

Comment:
It's being used to throw an exception in case data doesn't fit in. Instead of checking each element if it fits, I simply sort and select the biggest one. Granted I could just do `max(data)` though